src/sklearn_evaluation/models.py

```diff
@@ -0,0 +1,430 @@
+
+        # TODO: Evaluate model by confusion matrix
+        diagonal = np.diagonal(cm)  # noqa
+
+    def evaluate_auc(self, y_true, y_score):
+        """
+        Checks if roc auc is in acceptable range
+        """
+        auc_section = ReportSection("auc")
+
+        auc_threshold_low_range = Range(0, 0.6)
+        auc_threshold_acceptable_range = Range(0.7, 0.8)
+
+        # auc - roc
+        roc = plot.ROC.from_raw_data(y_true, y_score)
+        for i in range(len(roc.fpr)):
+            roc_auc = auc(roc.fpr[i], roc.tpr[i])
+
+            # todo: better check
+            label = roc.label[i] if len(roc.label) > 0 else f"class {i}"
+            r = re.match(r"^\(class (.)*\)", label)
+            if r:
+                class_name = r[0].replace("(", "").replace(")", "")
+            else:
+                class_name = label
+
+            if auc_threshold_low_range.in_range(roc_auc):
+                auc_section.append_guideline(
+                    f"Area under curve is low for {class_name}"
+                )
+                class_roc = plot.ROC(roc.fpr[i], roc.tpr[i], label=[label]).plot().ax
+                auc_section.append_guideline(class_roc)
+
+                auc_section.append_guideline(
+                    "To tackle this, check out this "
+                    "<a href='https://ploomber.io/blog/' target='_blank'>guide</a>"
+                )
+            elif auc_threshold_acceptable_range.in_range(roc_auc):
+                auc_section.set_is_ok(True)
+                auc_section.set_include_in_report(False)
+            else:
+                auc_section.set_is_ok(True)
+                auc_section.set_include_in_report(False)
+
+        self._add_section_to_report(auc_section)
+
+    def generate_general_stats(self, y_true, y_pred, y_score):
+        """
+        Include general stats in the report
+        """
+        general_section = ReportSection("general_stats")
+
+        general_section.append_guideline(
+            plot.confusion_matrix(y_true, y_pred, ax=_gen_ax())
+        )
+        general_section.append_guideline(plot.roc(y_true, y_score, ax=_gen_ax()))
+        self._add_section_to_report(general_section)
+
+
+class ModelComparer(ModelHeuristics):
+    """
+    Compares models and generate report
+    """
+
+    def __init__(self, model_a, model_b):
+        self.model_a = model_a
+        self.model_b = model_b
+        super().__init__()
+
+    def precision_and_recall(self, X_test, y_true):
+        """
+        Calculates precision and recall for each of the models
+        """
+        percision_recall_section = ReportSection("percision_recall")
+        pr_a = None
+        pr_b = None
+
+        try:
+            y_prob_a = self.model_a.predict_proba(X_test)
+            pr_a = plot.PrecisionRecall.from_raw_data(y_true, y_prob_a)
+
+            percision_recall_section.append_guideline(pr_a.ax_)
+
+        except Exception as exc:
```

Comment:
Putting a generic `Exception` isn't recommended; it's better to put the specific exception we want to cover and only have a single line in the try body. what could it fail here? I'm guessing the model might not have predict_proba?

the same comment applies to other `try catch` statements below